src/utilities/node-dimensions.ts

```diff
@@ -0,0 +1,21 @@
+import type { BaseNode } from '@/types';
+
+import {
+  DEFAULT_FIELD_HEIGHT,
+  DEFAULT_FIELD_PADDING,
+  DEFAULT_NODE_HEADER_HEIGHT,
+  DEFAULT_NODE_WIDTH,
+} from './constants';
+
+export const getNodeHeight = <N extends BaseNode>(node: N) => {
+  if ('height' in node && typeof node.height === 'number') return node.height;
+  if ('measured' in node && node.measured?.height) return node.measured.height;
+  const fieldCount = !('fields' in node) || !Array.isArray(node.fields) ? 1 : node.fields.length;
```

Comment:
Super nit, but I think just isArray check here should be enough (both from runtime and compiler perspective). Reads a bit better I think

```suggestion
  const fieldCount = Array.isArray(node.fields) ? node.fields.length : 1
```